### PR TITLE
test: add comprehensive test coverage for InviteService

### PR DESCRIPTION
**Pull request by gorkem-bwl**
## Summary
• Add comprehensive unit tests for InviteService business logic layer
• Cover all public methods: getInviteToken, sendInviteEmail, verifyInviteToken
• Include thorough error handling and edge case testing

This addresses a gap in backend test coverage identified during the test infrastructure analysis. The InviteService handles critical user invitation workflows including token generation, email sending, and verification.

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive unit tests for the invitation workflow, covering token generation, link construction, email composition/sending, verification, and multiple failure scenarios (SMTP/settings issues, template/build problems, and error propagation).
  * Improves coverage, reliability, and safeguards against regressions in onboarding invitations.
  * No user-facing changes.

### CODE REVIEW

File: server/tests/services/inviteService.test.js [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+			expect(inviteService.serviceName).to.equal("inviteService");
+		});
+	});
+
+	describe("static SERVICE_NAME", function () {
+		it("should have correct static service name", function () {
+			expect(InviteService.SERVICE_NAME).to.equal("inviteService");
+		});
+	});
+});
```
[the remark this answers is not in the document]
Comment (yunmin-nam) [reply server/tests/services/inviteService.test.js:253]: Resolved in c4f3d762: post-command readiness runs only for up/start/restart. stop returns after Compose stop without waiting for the stopped service; regression added.
